src/SettingsView/EntityMappingListItem.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import ConnectionComponent from './../Utils/ConnectionComponent.js';
+import config from './../config/config.js';
+import { Row, Col } from 'react-grid-system';
+import IconButton from 'material-ui/IconButton';
+import IconEdit from 'material-ui/svg-icons/editor/mode-edit';
+import IconDelete from 'material-ui/svg-icons/action/delete';
+import ListItem from 'material-ui/List';
+import {PromiseState} from 'react-refetch';
+
+class EntityMappingListItem extends ConnectionComponent {
+
+    constructor() {
+        super();
+        this.getEntityTypeName = this.getEntityTypeName.bind(this);
+        this.getEntityTypeAttributeName = this.getEntityTypeAttributeName.bind(this);
+        this.getEventTypeAttributeName = this.getEventTypeAttributeName.bind(this);
+    }
+
+    getEntityTypeName(entityTypeId) {
+        let searchedEntityType = undefined;
+        window.hierarchy.forEach(function(layer) {
+            layer.forEach(function (entityType) {
+                if (entityType.Id === entityTypeId) {
+                    searchedEntityType = entityType;
+                }
+            })
+        });
+        return searchedEntityType.Name;
+    }
+
+    getEntityTypeAttributeName(attributeId) {
+        const searchedAttribute = this.props.entityTypeAttributes.value.find(attribute => {
+            return attribute.Id === attributeId;
+        });
+        return searchedAttribute.Name;
+    }
+
+    getEventTypeAttributeName(attributeId) {
+        const searchedAttribute = this.props.eventTypeAttributes.find(attribute => {
+            return attribute.Id === attributeId;
+        });
+        return searchedAttribute.Name;
+    }
+
+    render() {
+        const allFetches = PromiseState.all([this.props.entityTypeAttributes]);
+        const connectionIncomplete = super.render(allFetches);
+        if(connectionIncomplete) {
+            return connectionIncomplete;
+        }
+
+        return (
+            <ListItem>
+                <Row>
+                    <Col md={10}>
+                        {this.props.eventType.Name} - {this.getEntityTypeName(this.props.mapping.EntityTypeId)}
+                    </Col>
+                    <Col md={2}>
+                        {/*TODO add functionality*/}

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)